src/Storefront/Theme/ThemeCompiler.php

```diff
@@ -240,6 +252,22 @@ private function dumpVariables(array $config, string $salesChannelId): string
             }
 
             if ($data['type'] === 'media') {
+                // Resolve path if media id is given
+                if (Uuid::isValid($data['value'])) {
+                    /** @var MediaEntity|null $media */
+                    $media = $this->mediaRepository
+                        ->search(
+                            new Criteria([$data['value']]),
```

Comment:
You should try to do that in batch (fetch multiple media elements at once)

Reply:
Thanks for the hint. Changed it in 88790bb to first gather the ids that needs resolving and then fetching them all in one batch.